components/sup/src/manager/service/config.rs

```diff
@@ -227,6 +234,18 @@ impl ServiceConfig {
         self.needs_write = false;
         Ok(should_restart)
     }
+
+    pub fn reload_package(&mut self, package: &PackageInstall) -> Result<()> {
+        self.config_root = match self.config_root_from_config {
+            Some(ref root) => root.clone(),
+            None => package.installed_path.clone(),
+        };
+        self.pkg = Pkg::new(package,
+                            self.pkg.svc_user.clone(),
+                            self.pkg.svc_group.clone())?;
+        self.cfg = Cfg::new(package, &self.config_root)?;
+        Ok(())
```

Comment:
Yeah I do exactly that in the service's update package right after this call. It did not seem right to mess with hooks inside the service config.